packages/@lwc/engine-core/src/framework/upgradable-element.ts

```diff
@@ -42,6 +44,21 @@ export function getUpgradableConstructor(
             }
         }
     };
+    if (features.ENABLE_NATIVE_CUSTOM_ELEMENT_LIFECYCLE) {
+        CE.prototype.connectedCallback = function () {
+            // FIXME: why is this undefined sometimes?
+            if (!isUndefined(getAssociatedVMIfPresent(this))) {
+                connectRootElement(this);
+            }
+        };
+
+        CE.prototype.disconnectedCallback = function () {
+            // FIXME: why is this undefined sometimes?
+            if (!isUndefined(getAssociatedVMIfPresent(this))) {
+                disconnectRootElement(this);
+            }
+        };
+    }
```

Comment:
These `FIXME`s are really worrisome. The `vm` seems to be undefined in some odd race conditions around slotting and portals, e.g. here:

https://github.com/salesforce/lwc/blob/fff5790097db66b8a05cdcf152ef270af624bcb9/packages/%40lwc/integration-karma/test/native-shadow/HTMLSlotElement-properties/x/NativeBasic/NativeBasic.js#L2-L7

Reply:
it could also be `undefined` if the constructor of the LWC Component throws during construction, in which case the instance this will never get the associated VM attached to it.

Reply:
I figured it out. The problem is that we're manually creating an element (`document.createElement('x-slotted')`) with a tag name already reserved by LWC. Added a check and a dev warning for that.